core/mutator/entry/mutation_test.go

```diff
@@ -42,3 +46,65 @@ func TestReplaceAuthorizedKeys(t *testing.T) {
 		}
 	}
 }
+
+func TestCreateAndVerify(t *testing.T) {
+	for _, tc := range []struct {
+		old     []byte
+		pubKeys []*keyspb.PublicKey
+		signers []signatures.Signer
+		data    []byte
+		wantErr bool
+	}{
+		{
+			old:     nil,
+			pubKeys: []*keyspb.PublicKey{mustPublicKey(testPubKey1)},
+			signers: []signatures.Signer{createSigner(t, testPrivKey1)},
+			data:    []byte("foo"),
+			wantErr: false,
+		},
+	} {
+		index := []byte{}
+		userID := "alice"
+		appID := "app1"
+
+		m, err := NewMutation(tc.old, index, userID, appID)
+		if err != nil {
+			t.Errorf("NewMutation(%v): %v", tc.old, err)
+			continue
+		}
+		if err := m.SetCommitment(tc.data); err != nil {
+			t.Errorf("SetCommitment(%v): %v", tc.data, err)
+			continue
+		}
+		err = m.ReplaceAuthorizedKeys(tc.pubKeys)
+		if got, want := err != nil, tc.wantErr; got != want {
+			t.Errorf("ReplaceAuthorizedKeys(%v): %v, wantErr: %v", tc.pubKeys, got, want)
+		}
+		update, err := m.SerializeAndSign(tc.signers)
+		if got, want := err != nil, tc.wantErr; got != want {
```

Comment:
This is the same wantErr as the one used to check the err from ReplaceAuthorizedKeys.
That seems to make it hard to test the case where ReplaceAuthorizedKeys returns nil but SerializeAndSign fails.

Reply:
this just just be `if err != nil`